airbyte-integrations/connectors/source-zendesk-chat/source_zendesk_chat/api.py

```diff
@@ -80,12 +73,117 @@ def get_stream_data(self, response_data: Any) -> List[dict]:
             raise Exception(f"Unsupported type of response data for stream {self.name}")
 
 
-class Agents(ZendeskChatStream):
+class ZendeskChatBaseIncrementalStream(ZendeskChatStream, ABC):
+    @property
+    @abstractmethod
+    def cursor_field(self) -> str:
+        """
+        Defining a cursor field indicates that a stream is incremental, so any incremental stream must extend this class
+        and define a cursor field.
+        """
+
+    @abstractmethod
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+
+    @staticmethod
+    def _field_to_datetime(value: Union[int, str]) -> pendulum.datetime:
+        if isinstance(value, int):
+            value = pendulum.from_timestamp(value / 1000.0)
+        elif isinstance(value, str):
+            value = pendulum.parse(value)
+        else:
+            raise ValueError(f"Unsupported type of datetime field {type(value)}")
+        return value
+
+
+class ZendeskChatTimeIncrementalStream(ZendeskChatBaseIncrementalStream, ABC):
```

Comment:
there is no need to mention Incremental in the name, also I think it is quite redundant to mention ZendeskChat, but it is up to you